### PR TITLE
Upgrade pgx and add go modules

### PR DESCRIPTION
**Pull request by gadelkareem**
- Upgrade pgx to v4
- Add go modules
- Add docker-compose for postgres
- Run test on Travis

### CODE REVIEW

**Comment by grippy** [comment]
Wish I saw this PR before working on a different issue 🤦  Didn't realize all the pgx changes required until I went to test my changes. Rather hold off until this is merged so I don't end up piling on here. Hopefully, the maintainer (@bgentry is that still you?)  has time to review this and leave feedback.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module github.com/gadelkareem/que
```
Comment (grippy) [inline go.mod:1]: This prefix prob needs to be changed. Curious, why the name change from `que-go` to `que`? Guessing this won't impact existing code still pointing to `github.com/bgentry/que-go`?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+vendor
```
Comment (grippy) [inline .gitignore:1]: Is this file needed?

**Comment by gadelkareem** [comment]
Oh sorry I thought you are not maintaining this repo anymore. Let me send you a new PR with the correct changes.
